Index: Bar.cc
===================================================================
RCS file: /mnt/back/repository/c/wbar/Bar.cc,v
retrieving revision 1.2
diff -u -r1.2 Bar.cc
--- Bar.cc	13 Jan 2007 20:57:03 -0000	1.2
+++ Bar.cc	13 Jan 2007 21:14:17 -0000
@@ -164,13 +164,14 @@
 /*}}}*/
 
 /* Icons Scaling *//*{{{*/
-void Bar::transform(int i_num, int i_off){ 
+void Bar::transform(int i_num, int i_off, float fact){ 
     int t_x, t_y, a;
 
     int x0 = (icon_ansd+1) * icon_unit,
 	xx = i_num * icon_unit + i_off,
 	rx = xx - icon_unit/2;
 
+    float tzoomf = 1.0 + (zoom_factor - 1.0) * fact;
     Icon *cur_ic=0;
 
     /* check wether we're over the icon *//*{{{*/
@@ -188,7 +189,7 @@
 	cur_ic = icons[a];
 
 	if( ((a<i_num)?i_num-a:a-i_num) > icon_ansd ){
-	    t_x = (int)(cur_ic->ox + ((a<i_num)?-1:1) * icon_ansd * b_dd / 2);
+	    t_x = (int)(cur_ic->ox + fact * ((a<i_num)?-1:1) * icon_ansd * b_dd / 2);
 	    t_y = cur_ic->oy;
 
 	    if(cur_ic->size != icon_size || cur_ic->x != t_x || cur_ic->y != t_y){
@@ -207,8 +208,8 @@
 	    cur_ic->csize = cur_ic->size;
 
 	    cur_ic->size = (int)( icon_size * 
-		( 1.0 + (zoom_factor-1.0)/(1.0 + expf(b_scl_a + b_scl_b*rx*rx))/b_scl_d));
-	    cur_ic->x = (int)( cur_ic->ox - icon_ansd * b_dd/2 + b_pos_m * (xx - x0) );
+		( 1.0 + (tzoomf-1.0)/(1.0 + expf(b_scl_a + b_scl_b*rx*rx))/b_scl_d));
+	    cur_ic->x = (int)( cur_ic->ox - fact * (icon_ansd * b_dd/2 - b_pos_m * (xx - x0)) );
 	    cur_ic->y = cur_ic->oy - (int)(jump_factor*(cur_ic->size-icon_size));
 	    cur_ic->need_update = 1;
 	}
@@ -335,18 +336,24 @@
 /*}}}*/
 
 /* Refresh *//*{{{*/
-void Bar::refresh(int mouse_x){
+void Bar::refresh(int mouse_x, int mouse_y){
     Icon *cur_ic;
 
     int iNum = iconNumber(mouse_x);
     int iOff = iconOffset(mouse_x);
+    //float iPer = (float)mouse_y / window->h;
+    /* Interpolated from: 
+     * 0    ->  0
+     * wh/2 ->  1 
+     * wh   ->  0 */
+    float iPer = 2.0 * mouse_y / window->h * (2.0 - 2.0 * mouse_y / window->h);
 
     /* on the bar */
     if(mouse_x > (int)(icon_offset + icon_size/2.0) && iNum < (int)icons.size()){
 
 	if(!focused) focus();
 
-	transform(iNum, iOff);
+	transform(iNum, iOff, iPer);
 #ifndef COOLMACHINE_N_SHITCODE
 	cleanBack();
 #endif
@@ -382,8 +389,8 @@
 inline void Bar::unfocus(){
     focused = 0;
 
-    x = (int)icon_offset;
-    width = (int)(window->w - 2*icon_offset);
+    //x = (int)icon_offset;
+    //width = (int)(window->w - 2*icon_offset);
     zoomed_icon = -1;
 
     drawBack();
@@ -392,8 +399,8 @@
 inline void Bar::focus(){
     focused = 1;
 
-    x=0;
-    width = window->w;
+    //x=0;
+    //width = window->w;
 
     drawBack();
 }
Index: Bar.h
===================================================================
RCS file: /mnt/back/repository/c/wbar/Bar.h,v
retrieving revision 1.2
diff -u -r1.2 Bar.h
--- Bar.h	13 Jan 2007 20:57:03 -0000	1.2
+++ Bar.h	13 Jan 2007 21:05:50 -0000
@@ -60,7 +60,7 @@
 
 	void iconPress(int i_num, int offs);
 
-	void transform(int num, int off);
+	void transform(int num, int off, float fact = 0.0);
 	virtual void render();
 
 	void acquireBack();
@@ -87,7 +87,7 @@
 	std::string iconCommand(int i_num);
 
 	/* Focus & unfocus events */
-	void refresh(int mouse_x = -1);
+	void refresh(int mouse_x = -1, int mouse_y = 0);
 
 	/* Icon press events */
 	void iconDown(int i_num);
Index: Main.cc
===================================================================
RCS file: /mnt/back/repository/c/wbar/Main.cc,v
retrieving revision 1.2
diff -u -r1.2 Main.cc
--- Main.cc	13 Jan 2007 20:57:03 -0000	1.2
+++ Main.cc	13 Jan 2007 21:06:48 -0000
@@ -228,9 +228,9 @@
 	    /* Motion Notify *//*{{{*/
 	    case MotionNotify:
 		if(!vertbar)
-		    barra->refresh(ev.xmotion.x);
+		    barra->refresh(ev.xmotion.x, ev.xmotion.y);
 		else
-		    barra->refresh(ev.xmotion.y);
+		    barra->refresh(ev.xmotion.y, ev.xmotion.x);
 		break;
 	    /*}}}*/
 
@@ -244,9 +244,9 @@
 	    case EnterNotify:
 		if(ev.xcrossing.mode!=NotifyUngrab && !(ev.xcrossing.state&Button1Mask)){
 		    if(!vertbar)
-			barra->refresh(ev.xcrossing.x);
+			barra->refresh(ev.xcrossing.x, ev.xcrossing.y);
 		    else
-			barra->refresh(ev.xcrossing.y);
+			barra->refresh(ev.xcrossing.y, ev.xcrossing.x);
 		}
 		break;
 	    /*}}}*/
Index: Makefile
===================================================================
RCS file: /mnt/back/repository/c/wbar/Makefile,v
retrieving revision 1.7
diff -u -r1.7 Makefile
--- Makefile	13 Jan 2007 20:57:03 -0000	1.7
+++ Makefile	13 Jan 2007 20:58:12 -0000
@@ -1,5 +1,5 @@
 TARGET=wbar
-CXXFLAGS=`imlib2-config --cflags` -Wall -O2 -DCOOLMACHINE_N_SHITCODE -DAVGFILTER -DDOCKWIN 
+CXXFLAGS=`imlib2-config --cflags` -Wall -O2 -DCOOLMACHINE_N_SHITCODE -DAVGFILTER #-DDOCKWIN 
 LDFLAGS=`imlib2-config --libs`  -Wl,-O2
 PREFIX=/usr/share/wbar
 
Index: SuperBar.cc
===================================================================
RCS file: /mnt/back/repository/c/wbar/SuperBar.cc,v
retrieving revision 1.3
diff -u -r1.3 SuperBar.cc
--- SuperBar.cc	13 Jan 2007 20:57:03 -0000	1.3
+++ SuperBar.cc	13 Jan 2007 21:07:17 -0000
@@ -147,8 +147,8 @@
 
     focused = 0;
 
-    x = (int)icon_offset;
-    width = (int)(window->w - 2*icon_offset);
+    //x = (int)icon_offset;
+    //width = (int)(window->w - 2*icon_offset);
     zoomed_icon = -1;
 
     /* Alfa filter the Bar */
