
--- Bar.cc	19 Oct 2006 23:25:01 -0000	1.1.1.1
+++ Bar.cc	22 Oct 2006 17:06:10 -0000
@@ -414,9 +414,10 @@
     /* on the bar */
     if(mouse_x > (int)(icon_offset + icon_size/2.0) && iNum < (int)icons.size()){
 
+	transform(iNum, iOff);
+
 	if(!focused) focus();
 
-	transform(iNum, iOff);
 #ifndef COOLMACHINE_N_SHITCODE
 	cleanBack();
 #endif
@@ -456,6 +457,30 @@
     width = (int)(window->w - 2*icon_offset);
     zoomed_icon = -1;
 
+#if 1
+    for(size_t i = 0; i<icons.size(); i++){
+	icons[i]->dx = (icons[i]->ox - icons[i]->x) / 15.0;
+	icons[i]->dy = (icons[i]->oy - icons[i]->y) / 15.0;
+	icons[i]->ds = (icon_size - icons[i]->size) / 15.0;
+	icons[i]->nx = icons[i]->x;
+	icons[i]->ny = icons[i]->y;
+	icons[i]->ns = icons[i]->size;
+    }
+
+    for(int i = 0; i < 15; i++){
+	for(size_t j = 0; j< icons.size(); j++){
+	    icons[j]->nx += icons[j]->dx;
+	    icons[j]->x = (int)icons[j]->nx;
+	    icons[j]->ny += icons[j]->dy;
+	    icons[j]->y = (int)icons[j]->ny;
+	    icons[j]->ns += icons[j]->ds;
+	    icons[j]->size = (int)icons[j]->ns;
+	}
+	render();
+    
+    }
+#endif
+
     drawBack();
 }
 
@@ -466,6 +491,29 @@
     width = window->w;
 
     drawBack();
+#if 1
+    for(size_t i = 0; i<icons.size(); i++){
+	icons[i]->dx = (icons[i]->x - icons[i]->ox) / 15.0;
+	icons[i]->dy = (icons[i]->y - icons[i]->oy) / 15.0;
+	icons[i]->ds = (icons[i]->size - icon_size) / 15.0;
+	icons[i]->nx = icons[i]->ox;
+	icons[i]->ny = icons[i]->oy;
+	icons[i]->ns = icon_size;
+    }
+
+    for(int i = 0; i < 15; i++){
+	for(size_t j = 0; j< icons.size(); j++){
+	    icons[j]->nx += icons[j]->dx;
+	    icons[j]->x = (int)icons[j]->nx;
+	    icons[j]->ny += icons[j]->dy;
+	    icons[j]->y = (int)icons[j]->ny;
+	    icons[j]->ns += icons[j]->ds;
+	    icons[j]->size = (int)icons[j]->ns;
+	}
+	render();
+    
+    }
+#endif
 }
 /*}}}*/
 

--- Icon.h	19 Oct 2006 23:25:01 -0000	1.1.1.1
+++ Icon.h	22 Oct 2006 17:06:12 -0000
@@ -18,6 +18,9 @@
 	int x, y;
 	int size;
 
+	float ds, dx, dy;
+	float nx, ns, ny;
+
 	/* redraw ? */
 	int need_update;
 	/* clean coords */

--- SuperBar.cc	19 Oct 2006 23:25:01 -0000	1.1.1.1
+++ SuperBar.cc	22 Oct 2006 17:06:15 -0000
@@ -132,6 +132,30 @@
 	FREE_IMAGE(bar);
 	bar = tBar;
     }
+
+#if 1
+    for(size_t i = 0; i<icons.size(); i++){
+	icons[i]->dx = (icons[i]->ox - icons[i]->x) / 10.0;
+	icons[i]->dy = (icons[i]->oy - icons[i]->y) / 10.0;
+	icons[i]->ds = (icon_size - icons[i]->size) / 10.0;
+	icons[i]->nx = icons[i]->x;
+	icons[i]->ny = icons[i]->y;
+	icons[i]->ns = icons[i]->size;
+    }
+
+    for(int i = 0; i < 10; i++){
+	for(size_t j = 0; j< icons.size(); j++){
+	    icons[j]->nx += icons[j]->dx;
+	    icons[j]->x = (int)icons[j]->nx;
+	    icons[j]->ny += icons[j]->dy;
+	    icons[j]->y = (int)icons[j]->ny;
+	    icons[j]->ns += icons[j]->ds;
+	    icons[j]->size = (int)icons[j]->ns;
+	}
+	render();
+    
+    }
+#endif
 }
 /*}}}*/
 
